.github/workflows/cargo-doc.yml

```diff
@@ -0,0 +1,55 @@
+name: Generate cargo docs
+on:
+  workflow_dispatch:
+    inputs: # null
+  push:
+    branches:
+      - gh-pages-automation
```

Comment:
This needs to be changed to `master` before merging.

Reply:
We could also do that on a different trigger (daily?), this one is just the obvious one. The whole build process takes ~20 minutes.